pages/api/user/play-history/index.ts

```diff
@@ -0,0 +1,97 @@
+import { ValidDate, ValidNumber, ValidObjectId } from '@root/helpers/apiWrapper';
+import isPro from '@root/helpers/isPro';
+import withAuth from '@root/lib/withAuth';
+import { PlayAttemptModel } from '@root/models/mongoose';
+import { LEVEL_DEFAULT_PROJECTION } from '@root/models/schemas/levelSchema';
+import { PipelineStage, Types } from 'mongoose';
+
+interface GetPlayAttemptsParams {
+  userId: Types.ObjectId;
+  cursor?: Types.ObjectId;
+  datetime?: Date;
+  minDurationMinutes?: number;
+}
+
+async function GetPlayAttempts(params: GetPlayAttemptsParams) {
```

Comment:
I haven't read the logic here in detail but there seems to be a bug with how the plays are grouped together. Most levels in my play history have 2+ consecutive entries (Aspirations in my recent plays has 21 consecutive entires!)